master/internal/rm/kubernetesrm/resource_pool.go

```diff
@@ -703,6 +703,15 @@ func (p k8sPodResources) Start(
 	spec.LoggingFields["task_id"] = spec.TaskID
 	spec.ExtraEnvVars[sproto.ResourcesTypeEnvVar] = string(sproto.ResourcesTypeK8sPod)
 	spec.ExtraEnvVars[resourcePoolEnvVar] = p.req.ResourcePool
+
+	// add custom labels to pods for observability
+	labels := make(map[string]string)
+	labels[userLabel] = spec.Owner.Username
+	labels[workspaceLabel] = spec.Workspace
+	labels[resourcePoolLabel] = p.req.ResourcePool
+	labels[taskTypeLabel] = string(spec.TaskType)
+	labels[taskIDLabel] = spec.TaskID
```

Comment:
I don't think we should add a Labels field to ```pod``` or ```StartTaskPod```

I think we should be able to just set these labels when are building the pod spec. The ```pod``` struct has the ```spec``` and the ```req``` that we are using to build this. The pod spec should be a source of truth with what labels the pod has.